core/src/main/java/com/linecorp/armeria/server/docs/DocService.java

```diff
@@ -257,34 +260,44 @@ static final class SpecificationLoader {
         }
 
         boolean contains(String path) {
-            return targetPaths.contains(path);
+            return TARGET_PATHS.contains(path);
         }
 
         CompletableFuture<List<AggregatedHttpFile>> updateServices(List<ServiceConfig> services,
                                                                    Executor executor) {
             this.services = services;
-            final List<CompletableFuture<AggregatedHttpFile>> files = targetPaths.stream().map(
-                    path -> load(path, executor)).collect(Collectors.toList());
-            return CompletableFutures.allAsList(files);
+
+            return generateServiceSpecification().thenCompose(
```

Comment:
Previously, the loading `CompletableFuture`s are executed synchronously and filled to `files` but now the futures will be set to `files` when `generateServiceSpecification()` is completed. It results in returning `loadFailedFuture` immediately instead of waiting for an `AggregatedHttpFile` to be completed.

Reply:
I have used @jrhee17 's solution and fired the future for `generateServiceSpecifications` first and assigned files without waiting for that future to complete.